### PR TITLE
Add content_filter hook

### PR DESCRIPTION
**Pull request by MrPetovan**
See: https://friendica.mrpetovan.com/display/735a2029715ab80d1874cf6808556656

Depends on https://github.com/friendica/friendica/pull/4722

Please review the translation strings I added, they may not be the best way of presening the content filtering reasons.

Additionally, the showmore addon now disabled when there is a content-filter applied because of display incompatibility.

### CODE REVIEW

**Comment by MrPetovan** [comment]
I'd like to wait until we agree on them as they currently are exact, but maybe too long or not friendly enough, or confusing maybe? On top of that I'm not sure about the procedure for addons.

**Comment by tobiasd** [comment]
In the addon directory you want to extract the strings `/path/to/run_xgettext.sh -s`

**Comment by MrPetovan** [comment]
I accidentally ran `/path/to/run_xgettext.sh -s` in the literal `/path/to/friendica/addon` folder, and I found an empty translation string that triggered a notice in gettext!

File: fromgplus/fromgplus.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ function fromgplus_addon_admin(&$a, &$o)
 
 	$o = replace_macros($t, [
 			'$submit' => L10n::t('Save Settings'),
-			'$key' => ['key', L10n::t('Key'), trim(Config::get('fromgplus', 'key')), L10n::t('')],
```
Comment (tobiasd) [inline fromgplus/fromgplus.php]: That is a really hard case for the translators ;-)

**Pull request merged**